### PR TITLE
feat: Add docker image build to workflows

### PR DESCRIPTION
**Pull request by marcpfuller**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [x] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
Adding a parallel work flow that builds all images 

## Issue this PR will close

close: #**issue_number**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+on: [pull_request]
+name: Build docker images
+jobs:
+  build_SOC:
+      runs-on: ubuntu-latest
+      steps:
+          - name: Checkout code
+              uses: actions/checkout@v3
+          - name: Make soc & dpgu images
```
Comment (jim-wang-intel) [inline .github/workflows/build.yaml:9]:
```suggestion
          - name: Make soc Docker images
```

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            run: |
+              make build-soc
+  build_DGPU:
+      runs-on: ubuntu-latest
+      steps:
+          - name: Checkout code
+              uses: actions/checkout@v3
+          - name: Make soc & dpgu images
```
Comment (jim-wang-intel) [inline .github/workflows/build.yaml:17]:
```suggestion
          - name: Make dpgu Docker images
```

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+              uses: actions/checkout@v3
+          - name: Build ovms client
+            run: |
+              make build-ovms-client
+  build_OVMS_SERVER:
+      runs-on: ubuntu-latest
+      steps:
+          - name: Checkout code
+              uses: actions/checkout@v3
+          - name: Build ovms client
```
Comment (jim-wang-intel) [inline .github/workflows/build.yaml:33]:
```suggestion
          - name: Build ovms server
```

**Review by brian-intel — APPROVED** [review]

**Review by vli11 — COMMENTED** [review]
it looks good if you can fix Jim and Brian's comments

**Review by brian-intel — APPROVED** [review]

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**